pydantic_ai_slim/pydantic_ai/models/openrouter.py

```diff
@@ -353,22 +359,78 @@ class OpenRouterChoice(chat_completion.Choice):
     Notably, removes 'function_call' and adds 'error'  finish reasons.
     """
 
-    message: OpenRouterCompletionMessage  # type: ignore[reportIncompatibleVariableOverride]
+    message: _OpenRouterCompletionMessage  # type: ignore[reportIncompatibleVariableOverride]
     """A wrapped chat completion message with OpenRouter specific attributes."""
 
 
-class OpenRouterChatCompletion(chat.ChatCompletion):
+class _OpenRouterCostDetails(BaseModel):
+    """OpenRouter specific cost details."""
+
+    upstream_inference_cost: int | None = None
+
+
+class _OpenRouterPromptTokenDetails(completion_usage.PromptTokensDetails):
+    """Wraps OpenAI completion token details with OpenRouter specific attributes."""
+
+    video_tokens: int | None = None
+
+
+class _OpenRouterCompletionTokenDetails(completion_usage.CompletionTokensDetails):
+    """Wraps OpenAI completion token details with OpenRouter specific attributes."""
+
+    image_tokens: int | None = None
+
+
+class _OpenRouterUsage(completion_usage.CompletionUsage):
+    """Wraps OpenAI completion usage with OpenRouter specific attributes."""
+
+    cost: float | None = None
+
+    cost_details: _OpenRouterCostDetails | None = None
+
+    is_byok: bool | None = None
+
+    prompt_tokens_details: _OpenRouterPromptTokenDetails | None = None  # type: ignore[reportIncompatibleVariableOverride]
+
+    completion_tokens_details: _OpenRouterCompletionTokenDetails | None = None  # type: ignore[reportIncompatibleVariableOverride]
+
+
+class _OpenRouterChatCompletion(chat.ChatCompletion):
     """Wraps OpenAI chat completion with OpenRouter specific attributes."""
 
     provider: str
     """The downstream provider that was used by OpenRouter."""
 
-    choices: list[OpenRouterChoice]  # type: ignore[reportIncompatibleVariableOverride]
+    choices: list[_OpenRouterChoice]  # type: ignore[reportIncompatibleVariableOverride]
     """A list of chat completion choices modified with OpenRouter specific attributes."""
 
-    error: OpenRouterError | None = None
+    error: _OpenRouterError | None = None
     """OpenRouter specific error attribute."""
 
+    usage: _OpenRouterUsage | None = None  # type: ignore[reportIncompatibleVariableOverride]
+    """OpenRouter specific usage attribute."""
+
+
+def _map_openrouter_provider_details(
+    response: _OpenRouterChatCompletion | _OpenRouterChatCompletionChunk,
+) -> dict[str, Any]:
+    provider_details: dict[str, Any] = {}
+
+    provider_details['downstream_provider'] = response.provider
+    provider_details['native_finish_reason'] = response.choices[0].native_finish_reason
```

Comment:
Our `provider_details['finish_reason']` is essentially the same thing as OpenRouter's `native_finish_reason`, as in the original value before being mapped to some abstraction layer. So it feels natural to me to call it `finish_reason` here. We'd "lose" OpenRouter's own intermediate value, but not really because that maps to the `response.finish_reason` values (I think).

I'm just thinking about code somewhere that may look for `provider_details['finish_reason']` and expect it to the the ultimate original value